builder/azure/common/vault.go

```diff
@@ -38,18 +43,17 @@ type SecretAttributes struct {
 
 func (client *VaultClient) GetSecret(vaultName, secretName string) (*Secret, error) {
 	p := map[string]interface{}{
-		"secret-name": secretName,
+		"secret-name": autorest.Encode("path", secretName),
```

Comment:
It's underbelly you should ever have to see, but it's exposed in this case because we do not have a KeyVault client.  I'll log a bug.